functions/Test-DbaDatabseBestPractice

```diff
@@ -0,0 +1,305 @@
+To return detailed information about databases and their best practice compliance 
+	
+#>
+	[CmdletBinding(SupportsShouldProcess = $true)]
+	Param (
+		[parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[Alias("ServerInstance", "SqlInstance")]
+		[string[]]$SqlServer,
+		[object]$SqlCredential,
+		[switch]$Detailed,
+		[switch]$IncludeSystemDbs #added this, saw you all talking about it
+	)
+	
+	DynamicParam { if ($SqlServer) { return Get-ParamSqlDatabases -SqlServer $SqlServer[0] -SqlCredential $Credential } }
+	
+	BEGIN
+	{
+		$databases = $psboundparameters.Databases
+		
+		$systemdbs = @("master", "msdb", "model", "tempdb")
+		
+		Function Test-AutoClose
+		{
+			switch ($db.AutoClose)
+			{
+				$true { return $false }
+				default { return $true }
+			}
+		}
+		
+		Function Test-AutoShrink
+		{
+			switch ($db.AutoShrink)
+			{
+				$true { return $false }
+				default { return $true }
+			}
+		}
+		
+		Function Test-PageVerify
+		{
+			if ($server.versionMajor -gt 8)
+			{
+				$bestpractice = "checksum"
+			}
+			else
+			{
+				$bestpractice = "tornpage"
+			}
+    			
+            switch ($db.PageVerify)
+			{
+				$bestpractice { return $true }
+				default { return $false }
+			}
+		}
+		
+		Function Test-AutoGrowth
+		{
+			$dbfiles = @()
+			
+			$filegroups = $db.FileGroups
+			
+			foreach ($filegroup in $filegroups)
+			{
+				foreach ($file in $filegroup.Files) 
+				{
+					$bpgrowthsize = $file.Size * 0.125
+					$isbestpractice = ($file.GrowthType -eq "Percent" -and $file.Growth -ge $bpgrowthsize)
+					
+					switch ($maxsize = $file.MaxSize)
+					{
+						-1 { $maxsize = "Unlimited" }
+					}
+					
+					$dbfiles += New-Object PSObject -Property @{
+						Server = $server.name
+						Database = $db.Name
+						FileGroupName = $filegroup.name
+						FileGroupType = $filegroup.FileGroupType
+						FileSize = $file.Size
+						GrowthType = $file.GrowthType
+						GrowthSize = $file.Growth
+						MaxSize = $maxsize
+						FileType = "Data"
+						IsBestPractice = $isbestpractice
+					}
+				}
+			}
+			
+			foreach ($file in $db.LogFiles)
+			{
+				$bpgrowthsize = $file.Size * 0.125
+				$isbestpractice = ($file.GrowthType -eq "Percent" -and $file.Growth -ge $bpgrowthsize)
+				
+				if ($file.MaxSize -eq -1)
+				{
+					$maxsize = "Unlimited"
+				}
+				else
+				{
+					$maxsize = $file.MaxSize
+					$maxsize = "$maxsize KB"
+				}
+				
+				$dbfiles = New-Object PSObject -Property @{
+					Server = $server.name
+					Database = $db.Name
+					FileGroupName = $file.name
+					FileGroupType = "Not Applicable" 
+					FileSize = $file.Size
+					GrowthType = $file.GrowthType
+					GrowthSize = $file.Growth
+					MaxSize = $maxsize
+					FileType = "Log"
+					IsBestPractice = $isbestpractice
+				}
+			}
+			
+			return $dbfiles
+		}
+
+		$collection = New-Object System.Collections.ArrayList
+	}
+	
+	PROCESS
+	{
+        $processed = New-Object System.Collections.ArrayList
+
+		foreach ($server in $sqlserver) 
```

Comment:
Doesn't accept multiple servers at the command line (creates a single server name from the whole list and tries to connect, failing each time. For example, server1,server2,server3 becomes server1 server2 server3 and results in the following error being displayed three times:

WARNING: Can't connect to server1 server2 server3 or access denied. Skipping.